### ISSUE TITLE
{% xlsmacro %} cannot call a macro defined in the same twig file

### ISSUE THREAD

**Issue by ThmX** [MewesK/TwigExcelBundle#8]
My goal is to call a macro from within the same file. I have the following code (using the [macro tag](http://twig.sensiolabs.org/doc/tags/macro.html) documentation):

```
{% xlsmacro test() %}
    {% xlsrow %}
        {% xlscell %}Test{% endxlscell %}
    {% endxlsrow %}
{% endxlsmacro %}

{% xlsmacro abc() %}
    {% from _self import test %}
    {% xlssheet 'Test' %}
        {% xlsrow %}
            {% xlscell %}Blabla{% endxlscell %}
        {% endxlsrow %}
        {{ test() }}
    {% endxlssheet %}
{% endxlsmacro %}
```

However I always end up with the following error:

```
[Tue Mar 29 04:38:37 2016] PHP Fatal error:  Call to a member function setRowIndex() on null in .../app/cache/dev/twig/fb/fb1e4a4bf3ba057f15e9a574aaa2995b54db470fe4d0d7b77082ab2ef10c9c36.php on line 48
[Tue Mar 29 04:38:37 2016] PHP Stack trace:
[Tue Mar 29 04:38:37 2016] PHP   1. {main}() .../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/router_dev.php:0
...
[Tue Mar 29 04:38:37 2016] PHP  13. Twig_Template->render($context = *uninitialized*) .../vendor/symfony/symfony/src/Symfony/Bridge/Twig/TwigEngine.php:50
[Tue Mar 29 04:38:37 2016] PHP  14. Twig_Template->display($context = *uninitialized*, $blocks = *uninitialized*) .../app/cache/dev/classes.php:4916
[Tue Mar 29 04:38:37 2016] PHP  15. Twig_Template->displayWithErrorHandling($context = *uninitialized*, $blocks = *uninitialized*) .../app/cache/dev/classes.php:4909
[Tue Mar 29 04:38:37 2016] PHP  16. __TwigTemplate_a7670c54b5cf726279d983da2ca013115b0363bfb24ee61994adf0137514face->doDisplay($context = *uninitialized*, $blocks = *uninitialized*) .../app/cache/dev/classes.php:4928
[Tue Mar 29 04:38:37 2016] PHP  17. __TwigTemplate_b6d34c3ce7c19ccfd8fda8227ebbb0e4f0f21490020b916bcb00bc28411f4dba->getreport($__date__ = *uninitialized*, $__report__ = *uninitialized*, $__phpExcel__ = *uninitialized*, ...$__varargs__ = variadic()) .../app/cache/dev/twig/e0/e06b7ba26da1610c2739834852c5c64d22ad463def1d9ee11e2e380c4d14b934.php:34
[Tue Mar 29 04:38:37 2016] PHP  18. __TwigTemplate_b6d34c3ce7c19ccfd8fda8227ebbb0e4f0f21490020b916bcb00bc28411f4dba->gettest($__phpExcel__ = *uninitialized*, ...$__varargs__ = variadic()) .../app/cache/dev/twig/fb/fb1e4a4bf3ba057f15e9a574aaa2995b54db470fe4d0d7b77082ab2ef10c9c36.php:122
```

**Comment by ThmX** [comment]
After debugging a bit I found out what is the problem. The fix is to pass the `_context.phpExcel` when calling the macro:

```
{{ test(_context.phpExcel) }}
```

But it's a momentarily hack as it's really an ugly fix.

**Comment by MewesK** [comment]
I'll try to have a proper fix ready by the end of the week. Thanks for your feedback!

**Comment by ThmX** [comment]
Fixed with #9 (I also added a test for it)

**Comment by MewesK** [comment]
Thanks for PR, looks good. Will be released soon as 2.1.
